EIPS/eip-3041.md

```diff
@@ -0,0 +1,95 @@
+---
+eip: 3041
+title: Add basefee in eth_getBlockByHash response
+author: Abdelhamid Bakhta (@abdelhamidbakhta)
+discussions-to: https://ethereum-magicians.org/t/eip-3041-add-basefee-in-eth-getblockbyhash-response/4825
+status: Draft
+type: Standards Track
+category : Interface
+created: 2020-10-13
+requires: 1559
+---
+
+## Simple Summary
+Add basefee field in `eth_getBlockByHash` RPC endpoint response.
+
+## Abstract
+[EIP-1559](/EIPS/eip-1559) introduces a base fee per gas in protocol.
+This value is maintained under consensus as a new field in the block header structure.
+This change should be reflected in RPC methods that return block header information.
+
+## Motivation
+Users may need value of the base fee at a given block. Base fee value is important to make gas price predictions more accurate.
+
+## Specification
+
+### eth_getBlockByHash
+
+#### Description
+
+Returns information about a block specified by hash.
+
+#### Parameters
+
+Parameters remain unchanged.
+
+#### Returns
+Add a new JSON field in the `result` object for block headers containing a base fee (post [EIP-1559](/EIPS/eip-1559) fork block).
+
+- {[`Quantity`](#quantity)} `baseFee` - base fee for this block
+
+#### Example
+
+```sh
+# Request
+curl -X POST --data '{
+    "id": 1559,
+    "jsonrpc": "2.0",
+    "method": "eth_getBlockByHash",
+    "params":["0xe670ec64341771606e55d6b4ca35a1a6b75ee3d5145a99d05921026d1527331", true]
+}' <url>
+
+# Response
+{
+    "id": 1559,
+    "jsonrpc": "2.0",
+    "result": {
+        "difficulty": "0x027f07",
+        "extraData": "0x0000000000000000000000000000000000000000000000000000000000000000",
+        "baseFee": "0x7"
+        "gasLimit": "0x9f759",
+        "gasUsed": "0x9f759",
+        "hash": "0xe670ec64341771606e55d6b4ca35a1a6b75ee3d5145a99d05921026d1527331",
+        "logsBloom": "0xe670ec64341771606e55d6b4ca35a1a6b75ee3d5145a99d05921026d1527331",
+        "miner": "0x4e65fda2159562a496f9f3522f89122a3088497a",
+        "nonce": "0xe04d296d2460cfb8472af2c5fd05b5a214109c25688d3704aed5484f9a7792f2",
+        "number": "0x1b4",
+        "parentHash": "0x9646252be9520f6e71339a8df9c55e4d7619deeb018d2a3f2d21fc165dde5eb5",
+        "sha3Uncles": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
+        "size":  "0x027f07",
+        "stateRoot": "0xd5855eb08b3387c0af375e9cdb6acfc05eb8f519e419b874b6ff2ffda7ed1dff",
+        "timestamp": "0x54e34e8e"
+        "totalDifficulty":  "0x027f07",
+        "transactions": []
+        "transactionsRoot": "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
+        "uncles": []
+    }
+}
+```
+---
+
+## Backwards Compatibility
+Backwards compatible. Calls related to block prior to [EIP-1559](/EIPS/eip-1559) fork block will omit the base fee field in the response.
```

Comment:
This should probably be part of the specification since interface changes are not necessarily in sync with consensus forks.  Something like this in the Specification section:
```
Every block returned by this endpoint whose block number is before the [EIP-1559](./eip-1559.md) fork block **MUST NOT** include a `baseFee` field.
Every block returned by this endpoint whose block number is on or after the [EIP-1559](./eip-1559.md) fork block **MUST** include a `baseFee` field.
```
It may still be worth mentioning this in the backward compatibility section, as it *is* related to backward compatibility, even if it is also explicitly part of the specification.

Reply:
Sounds good.